doc/cpp_simulation.rst

```diff
@@ -0,0 +1,139 @@
+        node.num_cpu_cores = arb::threading::num_threads();
+        node.num_gpus = arb::hw::num_gpus()>0? 1: 0; // use 1 GPU if any available
+
+        // Make a recipe of user defined type my_recipe.
+        my_recipe recipe;
+
+        // Get a description of the partition the model over the cores
+        // (and gpu if available) on node.
+        arb::domain_decomposition decomp = arb::partition_load_balance(recipe, node);
+
+        // Instatitate the simulation.
+        arb::simulation sim(recipe, decomp);
+
+
+Class Documentation
+-------------------
+
+.. cpp:namespace:: arb
+
+.. cpp:class:: simulation
+
+    The executable form of a model. A simulation is constructed
+    from a recipe, and then used to update and monitor model state.
+
+    Simulations take the following inputs:
+
+        * The **constructor** takes an :cpp:class:`arb::recipe` that describes
+          the model, and an :cpp:class:`arb::domain_decomposition` that
+          describes how the cells in the model are assigned to hardware resources.
+        * **Experimental inputs** that can change between model runs, such
+          as external spike trains.
+
+    Simulations provide an interface for executing and interacting with the model:
+
+        * **Advance model state** from one time to another and reset model
+          state to its original state before simulation was started.
+        * **I/O** interface for sampling simulation variables (e.g. compartment voltage
+          and current) and spike output.
+
+    **Types:**
+
+    .. cpp:type:: communicator_type = communication::communicator<communication::global_policy>
+
+        Type used for distributed communication of spikes and global synchronization.
+
+    .. cpp:type:: spike_export_function = std::function<void(const std::vector<spike>&)>
+
+        User-supplied callack function used as a sink for spikes generated
+        during a simulation. See :cpp:func:`set_local_spike_callback` and
+        :cpp:func:`set_global_spike_callback`.
+
+    **Constructor:**
+
+    .. cpp:function:: simulation(const recipe& rec, const domain_decomposition& decomp)
+
+    **Experimental inputs:**
+
+    .. cpp:function:: void inject_events(const pse_vector& events)
+
+        Add events directly to targets.
+        Must be called before calling :cpp:func:`simulation::run`, and must contain events that
+        are to be delivered at or after the current simulation time.
+
+    **Updating Model State:**
+
+    .. cpp:function:: void reset()
+
+        Reset the state of the simulation to its original state before
+        :cpp:func:`simulation::run` was called.
+
+    .. cpp:function:: time_type run(time_type tfinal, time_type dt)
+
+        Run the simulation from current simulation time to :cpp:var:`tfinal`,
+        with maximum time step size :cpp:var:`dt`.
+
+    .. cpp:function:: void set_binning_policy(binning_kind policy, time_type bin_interval)
+
+        Set event binning policy on all our groups.
+
```

Comment:
What a binning policy is is not described. Add a link to a placeholder or stub?

Reply:
I am inclined to leave it for now. The link will automatically be added when the docs for event binning are added.